lightkurve/correctors/pldcorrector.py

```diff
@@ -129,152 +262,543 @@ def correct(self, aperture_mask=None, cadence_mask=None, gp_timescale=30,
-            # long term approximation from over-fitting the transits.
-            XTX = np.dot(MX.T, MX)
-            XTX[np.diag_indices_from(XTX)] += 1e-8
-            XTy = np.dot(MX.T, M(rawflux))
-            y = M(rawflux) - np.dot(MX, np.linalg.solve(XTX, XTy))
-
-            # Estimate the amplitude parameter of a Matern-3/2 kernel GP
-            # by computing the standard deviation of y.
-            amp = np.nanstd(y)
-            tau = gp_timescale  # tau is a user-defined parameter
-            # set up gaussian process using celerite
-            # we use a Matern-3/2 kernel for its flexibility and non-periodicity
-            kernel = celerite.terms.Matern32Term(np.log(amp), np.log(tau))
-            gp = celerite.GP(kernel)
-            gp.compute(M(self.time), M(rawflux_err))
-
-            # compute the coefficients C on the basis vectors;
-            # the PLD design matrix will be dotted with C to solve for the noise model.
-            A = np.dot(MX.T, gp.apply_inverse(MX))
-            B = np.dot(MX.T, gp.apply_inverse(M(rawflux)[:, None])[:, 0])
+            cadence_mask = np.ones(len(self.raw_lc.time), dtype=bool)
+
+        # Theano raises an `AsTensorError` ("Cannot convert to TensorType")
+        # if the floats are not in 64-bit format, so we convert them here:
+        lc_time = np.asarray(self.raw_lc.time, np.float64)
+        lc_flux = np.asarray(self.raw_lc.flux, np.float64)
+        lc_flux_err = np.asarray(self.raw_lc.flux_err, np.float64)
+
+        # Covariance matrix diagonal
+        diag = lc_flux_err**2
+
+        # The cadence mask is applied by inflating the uncertainties in the covariance matrix;
+        # this is because celerite will run much faster if it is able to predict
+        # data for cadences that have been fed into the model.
+        diag[~cadence_mask] += 1e12
+
+        with pm.Model() as model:
+            # The mean baseline flux value for the star
+            mean = pm.Normal("mean", mu=np.nanmean(lc_flux), sd=np.std(lc_flux))
+            # Create a Gaussian Process to model the long-term stellar variability
+            # log(sigma) is the amplitude of variability, estimated from the raw flux scatter
+            logsigma = pm.Normal("logsigma", mu=np.log(np.std(lc_flux)), sd=5)
+            # log(rho) is the timescale of variability with a user-defined prior
+            logrho = pm.Normal("logrho", mu=np.log(gp_timescale_prior),
+                               sd=np.log(fractional_prior_width*gp_timescale_prior))
+            # Enforce that the scale of variability should be no shorter than 0.5 days
+            pm.Potential("logrho_prior", tt.switch(logrho > np.log(0.5), 0, np.inf))
+            # log(s2) is a jitter term to compensate for underestimated flux errors
+            # We estimate the magnitude of jitter from the CDPP (normalized to the flux)
+            s2_mu = self.raw_lc.estimate_cdpp() * 1e-6 * mean
```

Comment:
```suggestion
            s2_mu = self.raw_lc.estimate_cdpp() * 1e-6 * np.median(self.raw_lc.flux)
```

I'm finding that having the cyclic "`mean` is a distribution that also sets the distribution for `logs2`" hard to deal with, and my optimization seems better when the mean of `logs2` is just set to a value.